docs/release_process.rst

```diff
@@ -15,79 +14,112 @@ number occurs.
+As such, patch releases should only ever be based on an existing release point
+(git tag).
+
+This list assumes we are creating a new patch release off of the *latest*
+release version, i.e. off of the ``release`` branch.
+If a patch release for an older release version is being created, see the
+`Patching an Older Release`_ section.
+
+1. Create a new branch off of the ``release`` branch named something like
+   ``update-to-v{NEW_VERSION}``, where ``NEW_VERSION`` is the target ``X.Y.Z``,
+   including the bump in the patch (``Z``) version component.
 
-   a. Increment patch value in  ``pyproject.toml`` file's ``version`` attribute
-      under the `[tool.poetry]` section.
+   a. Use the ``scripts/update_release_notes.sh`` script to update the project
+      version number, create ``docs/release_notes/v{NEW_VERSION}.rst``, and add
+      a new pending release notes stub file. E.g.
 
-      * See `Poetry's version command`_ for a convenient means of incrementing
-        the version.
+      .. code-block:: bash
 
-   b. Rename the ``docs/release_notes/pending_release.rst`` file to
-      ``docs/release_notes/v{VERSION}.rst``, matching the new version value.
-      Add a descriptive paragraph under the title section summarizing this
-      release.
+         $ ./scripts/update_release_notes.sh patch
 
-   c. Add a reference to the new release notes RST file in
-      ``docs/release_notes.rst``.
+   b. Add a descriptive paragraph under the title section of
+      ``docs/release_notes/v{NEW_VERSION}.rst`` summarizing this release.
 
-   d. In a separate commit, add back a blank pending release notes file stub.
-      See `Stub Pending Notes File`_.
+2. Push the created branch to the upstream repository, not your fork (this is
+   an exception to the normal forky workflow).
 
-2. Create a pull/merge request for this branch with master as the merge target.
-   This is to ensure that everything passes CI testing before making the
-   release. If there is an issue, then branches should be made and merged into
-   this branch until the issue is resolved.
+3. Create a pull/merge request for this branch with ``release`` as the merge
+   target. This is to ensure that everything passes CI testing before making
+   the release. If there is an issue, then topic branches should be made and
+   merged into this branch until the issue is resolved.
 
-3. Tag branch (see `Tag new version`_ below) after resolving issues and before
-   merging into ``master``.
+4. Get a positive review.
 
-4. Merge version bump branch into the ``master`` branch.
+5. Merge the pull/merge request into the ``release`` branch.
 
-5. `Create new version release to PYPI`_.
+6. Tag the resulting merge commit.
+   See `Tag new version`_ below for how to do this.
+
+7. As a repository administrator, merge the ``release`` branch into ``master``
+   locally and push the updated ``master`` to upstream. (Replace "upstream"
+   with your applicable remote name.)
```

Comment:
Slightly different than the instructions in the major/minor release above, not sure it matters much:
```suggestion
   locally and push the updated ``master`` to upstream. (Replace "upstream"
   in the example below with your applicable remote name.)
```

Reply:
Good catch, I missed making that edit symmetric.